webanno-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/model/AnnotatorStateImpl.java

```diff
@@ -163,10 +165,17 @@
     
     private List<Unit> visibleUnits;
     
+    private transient ApplicationEventPublisher eventPublisher;
+    
     public AnnotatorStateImpl(Mode aMode)
     {
         mode = aMode;
     }
+    
+    public AnnotatorStateImpl(Mode aMode, ApplicationEventPublisher aEventPublisher) {
```

Comment:
You can't keep the ApplicationEventPublisher in the state - it is not serializable.

Reply:
I made it transient.
Should I avoid doing this?

Reply:
Yes. The state object is a data object. I think it shouldn't go around sending events. Also, I think we don't need to send a Spring event via the ApplicationEventPublisher - a Wicket event should suffice. Cf. check e.g. how `RenderAnnotationsEvent` is sent and recieved (via `@OnEvent`).

Reply:
With Wicket events, you could even send the event, cf. how `HybridApplicationUIEventRouter` fetches the current Page (if one is available) and then uses it to send an event.

Reply:
Alright, thanks I'll have a look.
But from where should these events be sent then?
From the editors (`LinkFeatureEditor`, `SubjectObjectFeatureEditor`, etc.)?
I added this to the `AnnotatorState` because it is the place where a slot is armed and it would ensure that a `SlotArmedEvent` is always sent.
If we do it from the editors we'd need to add it everywhere where `setArmedSlot` ist called, right?

Reply:
I changed it now for the `FeatureLinkEditor`.
Is this the way it should work?

Reply:
I edited my comment a couple of times. I think it should be ok to embed your `sendRenderSlotsEvent()` methods into `setArmedSlot()` - at least that's better than having to fire the events manually every time `setArmedSlot()` is called.